# Sprint 16: Add item requests and gateway

VitalyVL1 opened the pull request.

VitalyVL1 committed:
Добавить функционал создания запроса нужной вещи

VitalyVL1 committed:
Оптимизировать BookingServiceImpl и ItemRequestServiceImpl

VitalyVL1 committed:
Перед разделением на модули, после настроек, после разделения нужно будет удалить верификацию данных из POM сервера

VitalyVL1 committed:
Промежуточный вариант после разделения на модули

VitalyVL1 committed:
Рабочий вариант, до написания тестов.

VitalyVL1 committed:
Рабочий вариант, до написания тестов.

VitalyVL1 committed:
Временное отключение проверок на покрытие кода тестами для проверки прохождения тестов Postman, включить после покрытия кода тестами

VitalyVL1 committed:
Включить обратно проверки покрытия кода тестами.

VitalyVL1 committed:
Тестами покрыты DTO (Сериализация, Десериализация, Валидация) в модуле gateway (опционально)

VitalyVL1 committed:
Поправлена конфигурация POM для gateway

VitalyVL1 committed:
Написаны тесты для контроллеров в модуле server

VitalyVL1 committed:
Добавить поле с entityId в NotFoundException

VitalyVL1 committed:
Поправить опечатку в поле requestorID (requestorId) в ItemRequestResponseDto

VitalyVL1 committed:
Убрать не используемый конструктор из DuplicatedDataException

VitalyVL1 committed:
Убрать Обработку валидации из ExceptionHandlerController в Server

VitalyVL1 committed:
Добавить тесты для Service слоя

VitalyVL1 committed:
Code Style

VitalyVL1 committed:
Поправить валидацию в контроллерах,
Добавить обработку IllegalArgumentException

VitalyVL1 committed:
code style

VitalyVL1 committed:
Добавить тесты контроллеров и клиентов в gateway модуль

VitalyVL1 committed:
Добавить тесты Entity в server модуль

VitalyVL1 committed:
Добавить тесты ExceptionHandlerController в server модуль,
Убрать не используемые в модуле классы ValidationErrorResponse.java
Violation.java

Alexander-Bolotov commented on gateway/src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -0,0 +1,75 @@
+    private final ItemClient itemClient;
+
+    @PostMapping
+    public ResponseEntity<Object> addItem(
```
Типизировать Object не самый лучший вариант

Alexander-Bolotov commented on gateway/src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    @GetMapping
+    public ResponseEntity<Object> getItemsByOwner(
```
«…нужно, чтобы владелец видел даты последнего и ближайшего следующего бронирования для каждой вещи, когда просматривает список (GET /items).»

Alexander-Bolotov commented on server/src/main/resources/schema.sql:
```diff
@@ -15,7 +15,7 @@ CREATE TABLE IF NOT EXISTS users
 CREATE TABLE IF NOT EXISTS requests
```
schema.sql содержит DROP-ы таблиц — это ломает «многократный запуск»

VitalyVL1 committed:
Убрать DROP TABLE из schema.sql
Добавить видимость предыдущего и следующего бронирования вещи для владельца при просмотре списка своих вещей.
